scripts/dump-metrics-csv.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+import codecs
+import fnmatch
+import os
+import sys
+import yaml
+
+all_metrics = ""
+
+for root, dirs, files in os.walk("."):
+    for file in files:
+        if fnmatch.fnmatch(file, "metadata.yaml"):
+             with open(os.path.join(root, file)) as f:
+                data = yaml.load(f)
+                #print(data['monitors'])
+                mons = data['monitors']
+                if mons is None:
+                    continue
+                for m in mons:
+                    #print(m['monitorType'])
+                    if 'metrics' not in m.keys():
+                        continue
+                    metrics = m['metrics']
+                    if metrics is None:
+                        continue
+                    for mm in metrics:
+                        if 'description' not in metrics[mm].keys():
+                            desc = "THIS METRIC HAS NO DESCRIPTION"
+                        else:
+                            desc = str(metrics[mm]['description']).replace("\n", " ")
+                        if desc is None:
+                            desc = "THE METRIC DESCRIPTION IS BLANK"
+                        if 'default' not in metrics[mm].keys():
+                            default = False
+                        else:
+                            default = metrics[mm]['default']
+                        if default is None:
+                            default = "DEFAULT IS BLANK"
+                        if 'type' not in metrics[mm].keys():
+                            mtype = "THIS METRICS HAS NO TYPE"
+                        else:
+                            mtype = metrics[mm]['type']
+                        if mtype is None:
+                            mtype = "TYPE IS BLANK"
+                        all_metrics += ('"' + m['monitorType'] + '", "' + mm + '", "' + desc + '", "' + str(default) + '", "' + mtype + '"\n')
+output = codecs.open("all_metrics.csv", "w", "utf-8")
```

Comment:
codecs package is overkill here.  You just need `Path` from `pathlib`.  It can handle writing it out in UTF8.